modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library_settings/preview_limit.jsp

```diff
@@ -0,0 +1,102 @@
+%>
+
+<aui:form action="<%= dlPreviewLimitConfigurationDisplayContext.getEditDLPreviewLimitConfigurationURL() %>" method="post" name="fm">
+	<clay:sheet>
+		<liferay-ui:error exception="<%= ConfigurationModelListenerException.class %>" message="there-was-an-unknown-error" />
+
+		<liferay-ui:error exception="<%= DLPreviewLimitConfigurationException.InvalidMaxFileSizeException.class %>">
+			<liferay-ui:message key="maximum-file-size-limit-is-not-valid" />
+		</liferay-ui:error>
+
+		<liferay-ui:error exception="<%= DLPreviewLimitConfigurationException.InvalidMaxNumberOfPagesException.class %>">
+			<liferay-ui:message key="maximum-number-of-pages-limit-is-not-valid" />
+		</liferay-ui:error>
+
+		<clay:sheet-header>
+			<h2 class="c-mb-4">
+				<liferay-ui:message key="dl-preview-limit-configuration-name" />
+			</h2>
+		</clay:sheet-header>
+
+		<clay:sheet-section>
+			<h3 class="c-mb-2 sheet-subtitle text-2 text-secondary"><liferay-ui:message key="dl-preview-limit-configuration-maximum-file-size-title" /></h3>
+
+			<p class="c-mb-4 text-3 text-secondary">
+				<liferay-ui:message key="dl-preview-limit-configuration-maximum-file-size-help" />
+			</p>
+
+			<div>
+				<span aria-hidden="true" class="loading-animation"></span>
+
+				<react:component
+					module="document_library_settings/js/LimitedNumberInput"
+					props='<%=
+						HashMapBuilder.<String, Object>put(
+							"errorMessage", dlPreviewLimitConfigurationDisplayContext.getLimitExceededErrorMessage(dlPreviewLimitConfigurationDisplayContext.getMaxFileSizeLimit())
+						).put(
+							"label", LanguageUtil.get(resourceBundle, "maximum-file-size")
+						).put(
+							"limitValue", dlPreviewLimitConfigurationDisplayContext.getMaxFileSizeLimit()
+						).put(
+							"minimumValue", dlPreviewLimitConfigurationDisplayContext.getUnlimitedMaxFileSize()
+						).put(
+							"name", liferayPortletResponse.getNamespace() + "maxFileSize"
+						).put(
+							"unlimitedValue", dlPreviewLimitConfigurationDisplayContext.getUnlimitedMaxFileSize()
+						).put(
+							"value", dlPreviewLimitConfigurationDisplayContext.getMaxFileSize()
+						).build()
+					%>'
+				/>
+			</div>
+		</clay:sheet-section>
+
+		<clay:sheet-section>
+			<h3 class="c-mb-2 sheet-subtitle text-2 text-secondary"><liferay-ui:message key="dl-preview-limit-configuration-maximum-number-of-pages-title" /></h3>
+
+			<p class="c-mb-4 text-3 text-secondary">
+				<liferay-ui:message key="dl-preview-limit-configuration-maximum-number-of-pages-help" />
+			</p>
+
+			<div>
+				<span aria-hidden="true" class="loading-animation"></span>
+
+				<react:component
+					module="document_library_settings/js/LimitedNumberInput"
+					props='<%=
+						HashMapBuilder.<String, Object>put(
+							"errorMessage", dlPreviewLimitConfigurationDisplayContext.getLimitExceededErrorMessage(dlPreviewLimitConfigurationDisplayContext.getMaxNumberOfPagesLimit())
```

Comment:
We can inline this message inside the component. The only parameter needed is limitValue, so with the sub utility, we can create this message and do not need to pass it to the component: 
```
<ClayAlert 
     displayType="danger"  
     title="{sub(
              Liferay.Language.get('this-limit-is-higher-than-x-limit-enter-maximum-value-x'),
              limitValue
      )}"
     variant="feedback" 
/>
```

Reply:
Yes, that's a good proposal, however my point was the same: I wanted this React component to be reusable, by having a configurable `errorMessage`. Again, if you think the component cannot be reused then I would do the suggested refactoring.

Reply:
Also, if you think the component need some other change to be reusable just tell me.

Reply:
By default, we try to make easier components and only why needed, extend them to be used in other places. Do you have idea where can be this input used? 
Also, notice that this component is under document_library module. If you are thinking in a reusable component, should not be placed there.

Reply:
Finally I left the `errorMessage` as an input parameter (I've simplified the DisplayContext methods).
Your proposal would mean removing the `errorMessage` parameter but adding a new parameter for first X of the label `this-limit-is-higher-than-x-limit-enter-maximum-value-x`